src/pudl/analysis/epa_crosswalk.py

```diff
@@ -66,57 +67,42 @@ def _get_unique_keys(epacems: pd.DataFrame | dd.DataFrame) -> pd.DataFrame:
 def filter_crosswalk_by_epacems(
     crosswalk: pd.DataFrame, epacems: pd.DataFrame | dd.DataFrame
 ) -> pd.DataFrame:
-    """Inner join unique CEMS units with the EPA crosswalk.
+    """Inner join unique CEMS units with the epacamd_eia crosswalk.
 
     This is essentially an empirical filter on EPA units. Instead of filtering by
     construction/retirement dates in the crosswalk (thus assuming they are accurate),
     use the presence/absence of CEMS data to filter the units.
 
     Args:
-        crosswalk: the EPA crosswalk, as from pudl.output.epacems.epa_crosswalk()
+        crosswalk: epacamd_eia crosswalk
         unique_epacems_ids (pd.DataFrame): unique ids from _get_unique_keys
 
     Returns:
-        The inner join of the EPA crosswalk and unique epacems units. Adds
+        The inner join of the epacamd_eia crosswalk and unique epacems units. Adds
         the global ID column unit_id_epa.
 
     """
     unique_epacems_ids = _get_unique_keys(epacems)
     key_map = unique_epacems_ids.merge(
         crosswalk,
-        left_on=["plant_id_eia", "unitid"],
-        right_on=["CAMD_PLANT_ID", "CAMD_UNIT_ID"],
+        on=["plant_id_eia", "emissions_unit_id_epa"],
         how="inner",
     )
     return key_map
 
 
-def filter_out_unmatched(crosswalk: pd.DataFrame) -> pd.DataFrame:
-    """Remove unmatched or excluded (non-exporting) units.
-
-    Unmatched rows are limitations of the completeness of the EPA crosswalk itself, not of PUDL.
-
-    Args:
-        crosswalk (pd.DataFrame): the EPA crosswalk, as from pudl.output.epacems.epa_crosswalk()
-
-    Returns:
-        pd.DataFrame: the EPA crosswalk with unmatched units removed
-    """
-    bad = crosswalk["MATCH_TYPE_GEN"].isin({"CAMD Unmatched", "Manual CAMD Excluded"})
-    return crosswalk.loc[~bad].copy()
-
-
```

Comment:
Removed because the crosswalk in the PUDL db already filters out unmatched rows

Reply:
Is there some documentation of how the new pudl db version of the crosswalk has been transformed compared to the raw version?

Reply:
Not really, but I will add some to this PR. 
Basically what it does is 
- Change column names
- Remove calculable columns
- Remove leading zeros from numeric strings for the `generator_id` and `emissions_unit_id_epa` fields
- Drop NA fields for `plant_id_eia` (i.e., non matched)
- Conform dtypes